### PR TITLE
spring data jpa hw, mvn clean package done

### PR DESCRIPTION
**Pull request by vkucher49**

### CODE REVIEW

File: src/main/java/mate/academy/service/BookServiceImpl.java [hunk 1]
```diff
@@ -37,4 +37,21 @@ public BookDto findById(Long id) {
...
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book bookFromDb = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
+        bookFromDb.setAuthor(createBookRequestDto.getAuthor());
+        bookFromDb.setTitle(createBookRequestDto.getTitle());
+        bookFromDb.setDescription(createBookRequestDto.getDescription());
+        bookFromDb.setPrice(createBookRequestDto.getPrice());
+        bookFromDb.setCoverImage(createBookRequestDto.getCoverImage());
```
Comment (Rommelua) [inline src/main/java/mate/academy/service/BookServiceImpl.java:52]: check if ID exists and then use mapper

File: src/main/java/mate/academy/service/BookServiceImpl.java [hunk 2]
```diff
@@ -37,4 +37,24 @@ public BookDto findById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by ID: " + id));
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
+        if (id == null) {
+            throw new RuntimeException("ID do not exist: " + id);
```
Comment (koval-kn) [inline src/main/java/mate/academy/service/BookServiceImpl.java:44]:
```suggestion
            throw new RuntimeException("ID is null");
```

File: src/main/java/mate/academy/service/BookServiceImpl.java (hunk 1 above)
Comment (koval-kn) [reply src/main/java/mate/academy/service/BookServiceImpl.java:52]: mapper was not used

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

File: src/main/java/mate/academy/service/BookServiceImpl.java [hunk 3]
```diff
@@ -37,4 +37,24 @@ public BookDto findById(Long id) {
...
+        }
+        Book bookFromDb = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book by id " + id));
+        bookFromDb.setAuthor(createBookRequestDto.getAuthor());
+        bookFromDb.setTitle(createBookRequestDto.getTitle());
+        bookFromDb.setDescription(createBookRequestDto.getDescription());
+        bookFromDb.setPrice(createBookRequestDto.getPrice());
+        bookFromDb.setCoverImage(createBookRequestDto.getCoverImage());
```
Comment (sokimaaa) [inline src/main/java/mate/academy/service/BookServiceImpl.java:52]: use mapper here, e.g. mapstruct

File: src/main/java/mate/academy/controller/BookController.java [hunk 4]
```diff
@@ -33,4 +37,16 @@ public BookDto findBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto bookRequestDto) {
         return bookService.save(bookRequestDto);
     }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    void delete(@PathVariable Long id) {
```
Comment (sokimaaa) [inline src/main/java/mate/academy/controller/BookController.java:43]: never return void, consider returning ResponseEntity<?> with proper status code, and create dto for this, that shows delete was successfully finished or not

Comment (koval-kn) [reply src/main/java/mate/academy/controller/BookController.java:43]: not fixed

**Pull request merged**